src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -1239,13 +1238,20 @@ public void SetNonBlocking()
             //
             if (!_nonBlockingSet)
             {
-                if (Interop.Sys.Fcntl.SetIsNonBlocking(_socket, 1) != 0)
-                {
-                    throw new SocketException((int)SocketPal.GetSocketErrorForErrorCode(Interop.Sys.GetLastError()));
-                }
+                SetNonBlocking();
```

Comment:
This is put in a separate method for inlining?
Style wise using a local method may be nice. wdyt?

Reply:
Can't add NoInlining to a local method 😢